perfact/zodbsync/tests/test_sync.py

```diff
@@ -705,3 +705,278 @@ def test_extedit(self, encoding=None):
+
+    def test_watch_structure_changes_and_playback_local_changes(self, conn):
+        """
+        create structure while 'watch' command is running,
+        add local changes, then play those changes back and check,
+        if those changes played back correctly
+        """
+
+        # start watch daemon
+        watcher = self.mkrunner('watch')
+        watcher.setup()
+        app = conn.app
+        folder_1 = "folder_1"
+        s_folder_1 = "s_folder_1"
+
+        # create folder and wait until watch notices change
+        with conn.tm:
+            app.manage_addFolder(id=folder_1)
+        self.watcher_step_until(watcher,
+                                lambda: os.path.isdir(
+                                    self.repo.path + '/__root__/'+folder_1))
+
+        # create subfolder and wait until watch notices change
+        with conn.tm:
+            app.folder_1.manage_addFolder(id=s_folder_1, title=s_folder_1)
+        path = self.repo.path + '/__root__/'+folder_1+'/'+s_folder_1
+        self.watcher_step_until(watcher,
+                                lambda: os.path.isdir(path))
+
+        # change title
+        new_title = "new_title"
+        path = self.meta_file_path(folder_1, s_folder_1)
+        content = "[('title', '"+new_title+"'),('type', 'Folder'),]"
+        with open(path, 'w') as f:
+            f.write(content)
+
+        # playback changes and check if those are existent in zodb
+        self.run('playback', '/')
+        assert new_title == self.app.folder_1.s_folder_1.title
+
+        # wait for watch to notices played back changes
+        with open(path) as f:
+            meta = f.read()
+        self.watcher_step_until(watcher,
+                                lambda: "('title', '"+new_title+"')" in meta)
+
+    def test_commit_on_branch_and_exec_merge(self):
+        '''
+        change to a git feature branch and create a
+        structure there, commit it and change back to the master branch
+        on master branch check if changes from feature arent existent,
+        then merge feature branch and check if changes have been applied
+        correctly
+        '''
+
+        # change to feature branch and commit created folder/ subfolder
+        branch = "feature"
+        folder_1 = "folder_1"
+        s_folder_1 = "s_folder_1"
+        self.run('exec', 'git checkout -b {}'.format(branch))
+        self.app.manage_addFolder(id=folder_1)
+        self.app.folder_1.manage_addFolder(id=s_folder_1)
+        assert s_folder_1 in self.app.folder_1.objectIds()
+        self.run('record', '/')
+        assert os.path.isfile(self.meta_file_path(folder_1, s_folder_1))
+        self.gitrun('add', '-A')
+        self.gitrun('commit', '-m', 'test case 3')
+
+        # checkout to master and check that changes are not yet existent
+        self.run('exec', 'git checkout master')
+        assert os.path.isfile(self.meta_file_path(folder_1, s_folder_1)) \
+            is False
```

Comment:
According to a somewhat controversial recommendation from PEP8, you should not compare boolean values to `True` or `False` using `==` or `is` - I would agree with it unless you really need to semantically distinguish between `False` and other falsey values like `None`, which is not the case here. So it is more readable to simply write `assert not os.path.isfile(...)` rather than `assert os.path.isfile(...) is False`.

This occurs a couple of times in the code. Sorry I did not mention it in the first review.

Reply:
I refactored the code to your suggestion and I think it really looks cleaner now
thanks for the suggestion I will keep that style :)

Reply:
One argument for this is that if you compare a boolean to `True`, you get another boolean, which you could again compare - so where to stop?

```
if isfile:
    pass
if isfile == True:
    pass
if (isfile == True) == True:
    pass
if ((isfile == True) == True) == True:
    pass
```
So it is best to stop directly at the beginning, at least if it is clear that the value is a boolean. ;)